### DISCUSSION TITLE
Works sometimes, but keeps running into RuntimeError: Could not allocate tensor <...> bytes

### DISCUSSION THREAD

**Discussion by huangfanglong** [lshqqytiger/stable-diffusion-webui-directml#16] [Q&A] [answered]
Like the title said, most of my generation are met with 'not enough memory' issue, and generations take a long time (10-20 minutes or more).

GPU: 6900XT LC 16GB of VRAM GDDR6
Windows 11

I noticed when running the .bat it says: 
'Warning: caught exception 'Torch not compiled with CUDA enabled', memory monitor disabled'
No module 'xformers'. Proceeding without it.

I wonder if that matters or not but I figured it'd include it there. Initially I tried to do 1280x720 generations and it worked once then the subsequent generations keeps throwing the aforementioned Runtime Error, especially when it is close to completion.

I reduced the size to a square 500x500 with Hires.fix upscale to 800x800 and it still throws the error during generation and stops.

Batch count 2 and Batch size 1.

This is my current commandline args in webui-user.bat

COMMANDLINE_ARGS=--opt-sub-quad-attention --always-batch-cond-uncond --disable-nan-check --autolaunch

**Comment by morpheusseu** [comment]
as the error code said, AMD card has not CUDA hardware, which the xformers depands on. but it doesnt matter if you use AMD card for you will use directml instead.
with my 6600xt 8g, I have never used Hires.fix successfuly, recommend you upscale the real resolution and turn Hires.fix off, and my maxnium res is about 900*600.
tips: you could have some low res pictures and upscale it offline by deploying a [Upscayl](https://github.com/upscayl/upscayl) for free. but 
 after all a cheaper NVIDIA card will be much more effcient and easy to use than its higher level AMD card.

**Reply by huangfanglong** [reply]
Thanks for the tips, just started it today still messing around with a lot of these options.

**Comment by Miraihi** [comment, marked as answer]
I use [Ultimate SD upscale](url) extension to upscale the images I deem worthy. Works practically like hirez fix but with smaller chunks.
Your graphics card is a beast, my measily RX 580 does 1-2 s/iteration, less then a minute per picture. Try all these arguments:
--medvram --precision full --no-half --opt-sub-quad-attention --opt-split-attention-v1 --disable-nan-check

**Reply by morpheusseu** [reply]
the higher resolution the longer it cost, my 6600xt does 12min more on a 900600 pic (without hire.fix), but it's terrible when i get my new second hand card p102 10g, it run 1280720 pics with 1.5x Hires.fix into 19201080 with CUDA&xformers, which 4 batch size just cost 12min to generate (average 3 min per pic with 19201080 & 25 sample steps).
i hope AMD could develop a couterpart of NVIDIA's CUDA, which benifits a lot to AI calculation

**Reply by huangfanglong** [reply]
Thanks I'll try those.

EDIT: It's probably the combination of me stop using Hires.fix and your commandline arg that I am able to generate images much faster and consistently without the memory issue now, thanks guys.

**Reply by Enferlain** [reply]
> I use [Ultimate SD upscale](url) extension to upscale the images I deem worthy. Works practically like hirez fix but with smaller chunks. Your graphics card is a beast, my measily RX 580 does 1-2 s/iteration, less then a minute per picture. Try all these arguments: --medvram --precision full --no-half --opt-sub-quad-attention --opt-split-attention-v1 --disable-nan-check

What speeds are you getting? Used those arguments. I have a 6800xt and it shows like 30s/it for me on an 1024x1024 20step img2img and vram maxes out at 16gb and ram is at 25gb. I might as well be on cpu tbh

**Reply by Miraihi** [reply]
I can't go higher than 600x800. But even then, if I don't overload the prompt with lora, I get about 2s/it. 8Gb card. Maybe you won't need `--medvram`, but you probably need all other args: `--precision full --no-half --opt-sub-quad-attention --opt-split-attention`. For example, inpainting won't work at all without them.
Probably with your VRAM you can use the stock hirez fix no problem.
